cabal-install/src/Distribution/Client/Setup.hs

```diff
@@ -2230,11 +2230,9 @@ initOptions _ =
     (reqArg' "TOOL" (Flag . (:[]))
                     (fromFlagOrDefault []))
 
-    -- NB: this is a bit of a transitional hack and will likely be
-    -- removed again if `cabal init` is migrated to the v2-* command
-    -- framework
   , option "w" ["with-compiler"]
-    "give the path to a particular compiler"
+    "give the path to a particular compiler. For 'init', this flag is used \
+    \to set the base bound inference."
```

Comment:
Given that `init` is also for newbies, perhaps something like "the 'base' package lower bound inference"?

Reply:
sure that makes sense